### PR TITLE
Cleanup state, fix state sync and update issues

### LINKED ISSUE #33: Some state bugs throughout app not reseting/updating

### PR DESCRIPTION
**Pull request by 0xjimmy**
Fixes #33

### CODE REVIEW

**Comment by 0xjimmy** [comment]
Fixed state updates that didn't happen when settings were editied.
Updates in the transaction list to fetch abis and parse calldata that keeped getting called everyblock, no need to keep decoding the same calldata or making the same API requests.

Added button to attempt to fetch ABIs from Etherscan, users do not need to call if they do not want to parse the calldata or just not leak those contracts to Etherscan.

Made font sizes more consistent.

Gave overview of parameters used when calling simulate/submit.

image
image

The results from simulating, and status of submitted bundle attempts still needs to be cleaned up to be more readable. But the relay.dark.florist and relay-goerli.dark.florist were not responding today.
Will look at those in future pr, as well as error handling for when the relay RPCs are down.

File: app/ts/components/Submit.tsx [hunk 1]
```diff
@@ -229,18 +232,29 @@ export const Submit = ({
...
 			{missingRequirements.value ? (
 				<p>{missingRequirements.peek()}</p>
 			) : (
-				<div className='flex flex-col w-full gap-6'>
-					<Button onClick={simulateBundle} variant='secondary'>
-						Simulate
-					</Button>
+				<div className='flex flex-col w-full gap-4'>
+					<div>
+						<p><span className='font-bold'>Gas:</span> {utils.formatUnits(appSettings.value.priorityFee.toString(), 'gwei')} gwei + {utils.formatUnits(appSettings.value.priorityFee.toString(), 'gwei')} gwei priority</p>
```
Comment (KillariDev) [inline app/ts/components/Submit.tsx:248]:
```suggestion
						<p><span className='font-bold'>Gas:</span> {utils.formatUnits(appSettings.value.priorityFee.toString(), 'gwei')} nanoeth + {utils.formatUnits(appSettings.value.priorityFee.toString(), 'gwei')} nanoeth priority</p>
```

Comment (KillariDev) [reply app/ts/components/Submit.tsx:248]: we should use SI units

File: app/ts/components/Transactions.tsx [hunk 2]
```diff
@@ -82,13 +36,26 @@ export const Transactions = ({
...
 	const updateTx = async () => {
 		if (provider.value) {
 			const result = await createBundleTransactions(interceptorPayload.value, signers.value, blockInfo.value, appSettings.value.blocksInFuture, fundingAmountMin.value, provider.value.provider)
-			batch(() => {
+			if (Object.keys(interfaces.value).length === 0) {
 				transactions.value = result
-			})
+			} else {
+				const parsed = transactions.value.map((tx) => {
+					if (tx.transaction.to && tx.transaction.data && tx.transaction.data !== '0x' && tx.transaction.data.length > 0) {
```
Comment (KillariDev) [inline app/ts/components/Transactions.tsx:49]: here `tx.transaction.to` can be `0n` which is false?

Comment (KillariDev) [reply app/ts/components/Transactions.tsx:49]: with data, I guess you should also check for undefined too

Comment (KillariDev) [reply app/ts/components/Transactions.tsx:49]: tx.transaction?.data?.length > 0

File: app/ts/library/bundleUtils.ts [hunk 3]
```diff
@@ -106,7 +106,10 @@ export async function simulate(
 		blockInfo,
 		maxBaseFee,
 	)
-	return await flashbotsProvider.simulate(signedTransactions, Number(blockInfo.blockNumber + blocksInFuture))
+	const result = await flashbotsProvider.simulate(signedTransactions, Number(blockInfo.blockNumber + blocksInFuture))
+	console.log({ result })
```
Comment (KillariDev) [inline app/ts/library/bundleUtils.ts:110]: left out debug print?

**Review by KillariDev — APPROVED** [review]

File: app/ts/components/Submit.tsx (hunk 1 above)
Comment (0xjimmy) [reply app/ts/components/Submit.tsx:248]: Most common units people use for gas is gwei, nanoeth may be confusing for users if its not common on other apps.

**Pull request merged**